quinn-proto/src/connection/mod.rs

```diff
@@ -288,14 +288,8 @@ where
     /// - a call was made to `handle_event`
     /// - a call was made to `handle_timeout`
     pub fn poll(&mut self) -> Option<Event> {
-        if let Some(dir) =
-            Dir::iter().find(|&i| mem::replace(&mut self.streams.opened[i as usize], false))
-        {
-            return Some(Event::Stream(StreamEvent::Opened { dir }));
-        }
-
-        if let Some(id) = self.streams.poll_unblocked() {
-            return Some(Event::Stream(StreamEvent::Writable { id }));
+        if let Some(event) = self.streams.poll_events() {
```

Comment:
Nuance difference: this yields stream events before `Connected`/`ConnectionLost`/`Datagram` events. It passes the test suite and didn't seem like a big issue, please evaluate if it seems good enough.

Reply:
I think it's fine to order those three events last; with 0-RTT, stream events might happen before `Connected` regardless of how we handle the simultaneous case, and always emitting `ConnectionLost` after stream events is intuitive. It might be cute to have stream events arise before `Connected` iff they're from 0-RTT streams, but I don't think it's particularly necessary.